src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -591,6 +591,7 @@ class E2EHyperspaceRulesTests extends QueryTest with HyperspaceSuite {
   private def queryPlanHasExpectedRootPaths(
       optimizedPlan: LogicalPlan,
       expectedPaths: Seq[Path]): Boolean = {
+    assert(getAllRootPaths(optimizedPlan) === expectedPaths)
```

Comment:
Why do we need this check when the caller is doing assert? For printing diff using `===`?